src/bin/fullnode.rs

```diff
@@ -27,6 +32,7 @@ use std::sync::Arc;
 use std::thread::sleep;
 use std::time::Duration;
 
+#[cfg_attr(feature = "cargo-clippy", allow(cyclomatic_complexity))]
```

Comment:
Can we prevent this from sneaking in?  Or at least file a highpri issue to take it back out.   This kind of thing is pretty hard to eradicate and it'll only get worse over time

Reply:
👍 
I'll file a high priority issue

Reply:
Or just do it now? cyclomatic_complexity might be the world's best metric for "maintenance nightmare"

Reply:
https://github.com/solana-labs/solana/issues/2056

Reply:
@garious , sure let me see what can be done in this PR. I am hoping the current patch fixes the tests. The PR has grown a bit too big.